src/tools/analyzer/source_files.py

```diff
@@ -0,0 +1,58 @@
+from enum import Enum
+import configparser
+from os import listdir
+from os.path import isfile, join
+import glob
+
+
+class Reader:
+    class Component(Enum):
+        NSS = 1   # NSS Responder
+        PAM = 2   # PAM Responder
+        BE = 3    # Backend
+
+    def __init__(self, path):
+        self.log_files = []
+        self.path = self.resolve_path(path)
+        self.domains = self.get_domain_logfiles()
+
+    def __iter__(self):
+        for files in self.log_files:
+            try:
+                with open(files) as file:
+                    for line in file:
+                        yield line
+            except FileNotFoundError as err:
+                print("Could not find domain log file, skipping")
+                print(err)
+                continue
+
+    def resolve_path(self, path):
+        if path.endswith("/"):
+            return path
+        else:
+            return path + "/"
+
+    def get_domain_logfiles(self):
+        domain_files = []
+        exclude_list = ["ifp", "nss", "pam", "sudo", "autofs",
+                        "ssh", "pac", "kcm"]
+        file_list = glob.glob(self.path + "sssd_*")
+        for file in file_list:
+            if not any(s in file for s in exclude_list):
```

Comment:
I am wondering if it is worth to break this down a little for better visibility.

Reply:
I can do that, I do like how concise this expression is. I could also add a comment to clarify ?